rules/xcodeproj.bzl

```diff
@@ -117,29 +121,38 @@ def _xcodeproj_aspect_impl(target, ctx):
         providers.append(target_info)
     else:
         srcs = []
+        non_arc_srcs = []
         asset_srcs = []
-        other_cflags = []
         for attr in _dir(ctx.rule.files):
             if attr == "srcs":
                 srcs += getattr(ctx.rule.files, attr, [])
+            elif attr == "non_arc_srcs":
+                non_arc_srcs += getattr(ctx.rule.files, attr, [])
             else:
                 asset_srcs += getattr(ctx.rule.files, attr, [])
         srcs = [f for f in srcs if _is_current_project_file(f)]
+        non_arc_srcs = [f for f in non_arc_srcs if _is_current_project_file(f)]
         asset_srcs = [f for f in asset_srcs if _is_current_project_file(f)]
+        # if CcInfo in target:
+        #   for fi in target[CcInfo].framework_includes.framework_includes.to_list():
+        #     if fi[0] != '/':
+        #       fi = "$BAZEL_WORKSPACE_ROOT/%s" % fi
+        #     framework_includes.append("-F%s" % fi)
+        #   framework_includes.extend(["-D%s" % di for di in target[CcInfo].framework_includes.defines.to_list()])
+        #   framework_includes.append("-fobjc-arc") # TODO: only apply to arc sources
+        #   print("got other cflags %s" % framework_includes)
+        framework_includes = []
+        cc_defines = []
         if CcInfo in target:
-          for fi in target[CcInfo].compilation_context.framework_includes.to_list():
-            if fi[0] != '/':
-              fi = "$BAZEL_WORKSPACE_ROOT/%s" % fi
-            other_cflags.append("-F%s" % fi)
-          other_cflags.extend(["-D%s" % di for di in target[CcInfo].compilation_context.defines.to_list()])
-          other_cflags.append("-fobjc-arc") # TODO: only apply to arc sources
-          print("got other cflags %s" % other_cflags)
-
+          framework_includes = target[CcInfo].compilation_context.framework_includes.to_list()
```

Comment:
we shouldnt be calling `to_list` here, flattening depsets multiple times has a significant perf impact https://docs.bazel.build/versions/master/skylark/performance.html#avoid-calling-depsetto_list